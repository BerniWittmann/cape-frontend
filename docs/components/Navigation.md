# Navigation

Renders the Navigation Bar

