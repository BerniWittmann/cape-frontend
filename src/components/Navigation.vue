<template>
  <el-menu :default-active="this.activeRoute" mode="horizontal" :router="true" class="nav" active-text-color="#6CC9E8">
    <router-link
            :to="{ name: 'home' }"
            class="nav-image-wrap"
            tag="a"
    >
      <div class="nav-image"></div>
    </router-link>
    <el-menu-item :route="{ name: 'processes' }" index="processes">{{ $t('nav.processes') }}</el-menu-item>
    <el-menu-item :route="{ name: 'context_factors'}" index="context_factors">{{$t('nav.context_factors')}}</el-menu-item>
  </el-menu>
</template>
<script>
/* ============
 * Navigation Component
 * ============
 *
 * Navigation Bar component.
 *port DefaultLayout from '@/layouts/Default.vue'
 * Renders the navigation Bar.
 */

export default {
  computed: {
    activeRoute() {
      return this.$route.name
    }
  }
}

</script>

<style scoped lang="scss">
.nav-image-wrap {
  float: left;
  display: flex;
  align-items: center;
  text-decoration: none !important;
  outline: none !important;
  margin-right: 20px;

  .nav-image {
    margin: 10px 0 10px 20px;
    height: 40px;
    width: 100px;
    background: url('../assets/logo.png') no-repeat center center;
    background-size: contain;
  }
}

</style>
