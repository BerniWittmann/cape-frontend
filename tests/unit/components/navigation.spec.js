import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import { i18n } from '../setupPlugins'

import Navigation from '@/components/Navigation.vue'

describe('Components', () => {
  describe('Navigation', () => {
    it('renders', () => {
      const wrapper = shallowMount(Navigation, {
        i18n,
        stubs: {
          RouterLink: RouterLinkStub
        },
        mocks: {
          $route: {
            name: 'home'
          }
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('loads the correct route Name', () => {
      const wrapper = shallowMount(Navigation, {
        i18n,
        stubs: {
          RouterLink: RouterLinkStub
        },
        mocks: {
          $route: {
            name: 'home'
          }
        }
      })
      expect(wrapper.vm.activeRoute).toEqual('home')
    })
  })
})
