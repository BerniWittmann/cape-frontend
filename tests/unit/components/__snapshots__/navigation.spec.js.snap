// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Navigation renders 1`] = `
<elmenu-stub mode="horizontal" defaultactive="home" router="true" menutrigger="hover" activetextcolor="#6CC9E8" collapsetransition="true" class="nav"><a class="nav-image-wrap">
    <div class="nav-image"></div>
  </a>
  <elmenuitem-stub index="about" route="[object Object]">nav.about</elmenuitem-stub>
</elmenu-stub>
`;
